### DISCUSSION TITLE
getSession and rpc using different api keys

### DISCUSSION THREAD

**Discussion by magick93** [supabase/supabase#9752] [Questions]
After signing in, I would expect subsequent `rpc` calls to use the session (apikey) of the signed in user. This was happening as expected prior to upgrading to supabase-js v2.

The following is no longer using the logged in users apikey /session - instead it uses `anon`. 

The below is from a `cypress` test.
```
await supabase.auth
			.signInWithPassword({
				email: Cypress.env(issuerCredentials)['email'],
				password: Cypress.env(issuerCredentials)['password'],
			})
const { data:uSession } = await supabase.auth.getSession()
console.log(`uSession`, uSession.session)

const { data, error, status } = await supabase.rpc('add_claim_template', {
	title: faker.random.word(),
	description: faker.random.words(),
})
```

What am I doing wrong?

**Comment by kaleabmelkie** [comment]
A temporary workaround I found was overriding the `fetch` in the Supabase `createClient` gloabl options, and manually converting the `Headers` iterable type into an object.

That is:

```ts
import { PUBLIC_SUPABASE_PUBLIC_ANON_KEY, PUBLIC_SUPABASE_URL } from '$env/static/public'
import type { Database } from '$lib/types/supabase.generated'
import { createClient } from '@supabase/supabase-js'

export const supabase = createClient<Database, 'public'>(
  PUBLIC_SUPABASE_URL,
  PUBLIC_SUPABASE_PUBLIC_ANON_KEY,
  {
    global: {
      fetch: async (input, init) => {
        return fetch(input, {
          ...init,
          headers: {
            ...init?.headers,
            // eslint-disable-next-line
            // @ts-ignore
            ...Object.fromEntries(init?.headers?.entries?.() ?? new Map().entries()),
          },
        })
      },
    },
  },
)

```

I came to this solution because all supabase-js (v2) requests sent from Cypress:

1. Didn't attach the `Authorization` header
2. Attached unnecessary values such as `set`, `get`, `append`, `forEach`, `entries`, etc., which where the methods of a `Headers` object.

The supabase-js v2 request headers sent from Cypress looked like this before the above temporary patch:

<img width="683" alt="Screen Shot 2022-10-21 at 2 11 23 PM" src="https://user-images.githubusercontent.com/29814842/197308149-a6c518a4-5be6-4310-ac2e-e9e0c6d8c79a.png">
